### ISSUE TITLE
querySelector performance with attribute filter

### ISSUE THREAD

**Issue by skeggse** [tmpvar/jsdom#677]
I've found that `querySelector` and `querySelectorAll` perform well until you add attribute filters. For example:

```js
// performs "well"--a couple milliseconds
window.document.querySelectorAll('body > a[href]');
// performs very poorly (for a large (~1000) set of elements it can take many seconds)
window.document.querySelectorAll('body > a[href^=\'/resource\']');
```

What causes this? I can do without, but it doesn't seem like there should be quite such a large discrepancy between these two uses.

**Comment by dperini** [comment]
Probably fixed by the same commit referenced in issue #1490 ...
Now uses the jsdom own implementation of getAttribute/hasAttribute instead of the internals IE < 9 polyfills used in nwmatcher. I should definitely rewrite a jsdom only version of nwmatcher without the old IE bug-fixing cruft. This would allow for exceptional performances improvements.
In the mean time give it a try with the latest changes and let me know while I am working on it.

**Comment by domenic** [comment]
Confirmed fixed in the same way as #1490!
